### PR TITLE
refactor: Make network more reliable

### PR DESCRIPTION
**Pull request by TheWaWaR**
- [x] Change CKBProtocolHandler API
- [x] Change CKBProtocolContext API
- [x] Remove sqlite connection pool
- [x] Clean up PeerRegistry
- [x] Gracefully shutdown network
- [x] Prettify log

### CODE REVIEW

**Comment by TheWaWaR** [comment]
rebsaed

**Comment by TheWaWaR** [comment]
bors try

File: network/src/protocols/outbound_peer.rs [hunk 1]
```diff
@@ -51,21 +55,24 @@ impl OutboundPeerService {
...
+        let peers = self
+            .network_state
+            .with_peer_store(|peer_store| peer_store.peers_to_feeler(count));
+        let p2p_control = self.p2p_control.clone();
         for (peer_id, addr) in peers
             .into_iter()
             .filter(|(peer_id, _addr)| self.network_state.local_peer_id() != peer_id)
         {
+            self.network_state.with_peer_registry_mut(|reg| {
+                reg.add_feeler(peer_id.clone());
```
Comment (jjyr) [inline network/src/protocols/outbound_peer.rs]: What if dial failed?

PeerRegistry should only store peers which already connected, move this into 
 `Feeler.connected`.

Comment (jjyr) [inline network/src/protocols/outbound_peer.rs]: What if dial failed?

PeerRegistry should only store peers which already connected, move this into 
 `Feeler.connected`.

**Pull request merged**